driver-core/src/main/com/mongodb/operation/ChangeStreamOperation.java

```diff
@@ -205,38 +256,159 @@ public Collation getCollation() {
-                    callback.onResult(null, t);
+                    errHandlingCallback.onResult(null, t);
                 } else {
-                    callback.onResult(new AsyncChangeStreamBatchCursor<T>(ChangeStreamOperation.this, result, binding),
-                            null);
+                    final SingleResultCallback<AsyncBatchCursor<T>> wrappedCallback = releasingCallback(errHandlingCallback, source,
+                            connection);
+                    validateReadConcernAndCollation(source, connection, binding.getSessionContext().getReadConcern(), collation,
+                            new AsyncCallableWithConnectionAndSource() {
+                                @Override
+                                public void call(final AsyncConnectionSource source, final AsyncConnection connection, final Throwable t) {
+                                    if (t != null) {
+                                        wrappedCallback.onResult(null, t);
+                                    } else {
+                                        executeWrappedCommandProtocolAsync(binding, namespace.getDatabaseName(),
+                                                getCommand(connection.getDescription(), binding.getSessionContext()),
+                                                CommandResultDocumentCodec.create(RAW_BSON_DOCUMENT_CODEC, FIELD_NAMES_WITH_RESULT),
+                                                connection, asyncTransformer(source, connection),
+                                                new SingleResultCallback<AsyncBatchCursor<RawBsonDocument>>() {
+                                                    @Override
+                                                    public void onResult(final AsyncBatchCursor<RawBsonDocument> result,
+                                                                         final Throwable t) {
+                                                        if (t != null) {
+                                                            wrappedCallback.onResult(null, t);
+                                                        } else {
+                                                            wrappedCallback.onResult(
+                                                                    new AsyncChangeStreamBatchCursor<T>(ChangeStreamOperation.this,
+                                                                            result, binding), null);
+                                                        }
+                                                    }
+                                                });
+                                    }
+                                }
+                            });
                 }
             }
         });
     }
 
-    private AggregateOperation<RawBsonDocument> createAggregateOperation() {
+    private BsonDocument getCommand(final ConnectionDescription description, final SessionContext sessionContext) {
```

Comment:
Can no longer extend the aggregation option as we need the description to generate the command.